userspace/libsinsp/filter/parser.cpp

```diff
@@ -43,14 +43,73 @@ limitations under the License.
 	#include <regex>
 #endif  // _WIN32
 
+namespace
+{
+enum RGEX_ID
+{
+	RGX_NOTBLANK_ID,
+	RGX_IDENTIFIER_ID,
+	RGX_FIELDNAME_ID,
+	RGX_FIELDARGBARESTR_ID,
+	RGX_HEXNUM_ID,
+	RGX_NUMBER_ID,
+	RGX_BARESTR_ID
+};
+
+const std::string rgex_patterns[] = {
```

Comment:
Shouldn't the `USE_POSIX_REGEX` ifdef be pushed here? We need to support both standards, given our compiler compatibility coverage.